### PR TITLE
Update repository

### PR DESCRIPTION
**Pull request by junha1**

### CODE REVIEW

File: common/src/types.rs [hunk 1]
```diff
@@ -139,3 +139,10 @@ pub enum Commit {
     ExtraAgendaTransaction(ExtraAgendaTransaction),
     ChatLog(ChatLog),
 }
+
+/// The special finalization proof commit in the `pf` branch.
```
Comment (YoonhoKim0527) [inline common/src/types.rs:143]: at the below, it says `fp` branch. Is `fp` branch and `pf` branch is individual?

Comment (YoonhoKim0527) [inline common/src/types.rs:143]: needs to change pf to fp

File: repository/src/lib.rs [hunk 2]
```diff
@@ -103,9 +124,10 @@ impl<T: RawRepository> DistributedRepository<T> {
     /// Checks the validity of the repository, starting from the given height.
     ///
     /// It checks
-    /// 1. all the reserved branches and tags
+    /// 1. all the reserved branches and tags (especially valiity of the `fp` branch)
```
Comment (YoonhoKim0527) [inline repository/src/lib.rs:127]: What do you mean `especially` here? 
Do you wanted to say that we always have to check the validity of `fp` branch?

Comment (junha1) [reply repository/src/lib.rs]: It is just emphasized as it's newly added. I updated the description.

File: docs/git.md [hunk 3]
```diff
@@ -62,6 +62,7 @@ These are the names of the branches that are specially treated by the Simperby n
 3. `p`: the block proposal for this node. The node operator may push or force-push to this branch. When pushed, the Git server will check the validity of the branch. The consensus engine will recognize this branch and propose to the consensus. It stands for 'block proposal'.
 4. `a-<number>`: a valid agenda (but not yet approved) propagated from other nodes. If the governance has approved the agenda, it will point to the `agenda-proof` commit which lies on top of the agenda commit. The number is arbitrarily assigned.
 5. `b-<number>`: a valid (but not yet finalized) block propagated from other nodes. The number is arbitrarily assigned.
+6. `fp`: a very special branch that always holds the finalization proof for the last block. This is required because a block header doesn't contain the finalization proof of itself. Thus, to make a repository self-verifiable, it is essential to have the proof somewhere, in some way. This branch has only one empty commit that is directly on top of the genesis commit, titled with `finalization proof of the last block`. The commit message body contains the actual proof.
```
Comment (TomTaehoonKim) [inline docs/git.md:65]: Is empty commit a commit with no state transition? am I right?

File: repository/src/lib.rs [hunk 4]
```diff
@@ -52,26 +52,56 @@ impl<T: RawRepository> DistributedRepository<T> {
...
+    ///
+    /// It will leave only
+    /// - the `finalized` branch
+    /// - the `work` branch
+    /// - the `fp` branch.
+    ///
+    /// and
+    /// - the `p` branch
+    /// - the `a-#` branch
+    /// - the `b-#` branch
```
Comment (TomTaehoonKim) [inline repository/src/lib.rs:89]:
```suggestion
    /// - the `p` branches
    /// - the `a-#` branches
    /// - the `b-#` branches
```

**Review by JeongHunP — APPROVED** [review]

File: docs/git.md (hunk 3 above)
Comment (junha1) [reply docs/git.md:65]: an 'empty commit' is a term from Git (not our original context); a commit that does not have a diff, but only the commit message.

File: repository/src/lib.rs (hunk 4 above)
Comment (junha1) [reply repository/src/lib.rs:89]: note that `p` branch is unique. updated.

File: docs/git.md (hunk 3 above)
Comment (TomTaehoonKim) [reply docs/git.md:65]: understood

File: repository/src/lib.rs (hunk 4 above)
Comment (TomTaehoonKim) [reply repository/src/lib.rs]: oh right, understood

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Pull request merged**